### PR TITLE
Add unit test for nodes function

### PR DESCRIPTION
**Pull request by suzil**
- Additionally aggregates dummy agents into `runtests.jl`

Refs: #72

### CODE REVIEW

File: test/space_test.jl [hunk 1]
```diff
@@ -152,4 +152,23 @@ end
...
+  end
+  space = Space((3,3))
+  model = ABM(MyAgent, space)
+  for node in nodes(model)
+    if rand() > 0.7
+      add_agent!(node, model)
+    end
+  end
+  @test nodes(model) == 1:9
+  @test nodes(model, by=:random) == [8, 9, 1, 7, 6, 5, 3, 2, 4]
```
Comment (Datseris) [inline test/space_test.jl]: Shouldn't you set the seed here, to ensure that this random sequence will be the same all the time?

Comment (suzil) [reply test/space_test.jl]: There's a seed set at the beginning of the file.

File: test/runtests.jl [hunk 2]
```diff
@@ -1,5 +1,37 @@
...
+  weight::Float64
+end
+
+mutable struct Agent3 <: AbstractAgent
+  id::Int
+  pos::Tuple{Int,Int}
+  weight::Float64
+end
+
+mutable struct Agent4 <: AbstractAgent
```
Comment (suzil) [inline test/runtests.jl]: Maybe some of these could be further combined? I wasn't sure

Comment (Datseris) [reply test/runtests.jl]: Yeah, in total we need only 4 types of agents for the tests, the multiplicity of with and without position, and with and without some extra parameter `weight` or `a` or whatever.

Comment (Datseris) [reply test/runtests.jl]: Sorry, I was mistaken. You are correct we need 6 agents, because no position, integer position and tuple position are 3.

**Pull request merged**